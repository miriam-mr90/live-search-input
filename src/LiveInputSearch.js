import React, { Component } from 'react';

import View from './LiveInputSearch.view.jsx';

export default class LiveInputSearch extends Component {
    constructor(props) {
        super();

        this.state = {
            results: [],
            isFetching: false,
        };

        this.searchResults = this.searchResults.bind(this);
    }

    isResultsEmpty() {
        return this.state.results.length > 0;
    }

    isValueIncludeInText(value, text) {
        return text ? text.toLowerCase().indexOf(value.toLowerCase()) !== -1 : false;
    }

    searchResults(value) {
        this.setState({
            isFetching: true,
        });

        let resultsToDisplay = [];

        if(value !== '' && value.length > 1) {
            const availableResults = this.props.services;

            resultsToDisplay = availableResults.filter( item => (
                this.isValueIncludeInText(value, item.name) || this.isValueIncludeInText(value, item.parent_name)
            ));
        };

        this.setState({
            results: resultsToDisplay,
            isFetching: false,
        });
    }

    render() {
        return (
            <View
                {...this.props}
                {...this.state}
                displayResults={!!this.isResultsEmpty()}
                searchResults={this.searchResults}
            />
        );
    }
}
