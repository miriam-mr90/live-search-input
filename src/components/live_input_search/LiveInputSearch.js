import React, { Component } from 'react';
import PropTypes from 'prop-types';

import View from './LiveInputSearch.view.jsx';

export default class LiveInputSearch extends Component {
    constructor() {
        super();

        this.state = {
            results: [],
            isFetching: false,
            inputValue: null,
        };

        this.searchResults = this.searchResults.bind(this);
        this.getHighlightedItem = this.getHighlightedItem.bind(this);
    }

    isResultsEmpty() {
        return this.state.results.length > 0;
    }

    getHighlightedItem(value, item) {
        const year = `<span>${this.props.translations.from} ${item.Year}</span>`;
        let title = item.Title;
        const regex = new RegExp(`\\b${value}\\b`, 'i');
        const matches = title.match(regex);

        if (matches) {
            matches.forEach(m => (title = title.replace(m, `<em class="position">${m}</em>`)));
        }

        return `${title} ${year}`;
    }

    searchResults(value) {
        this.setState({
            results: [],
            inputValue: '',
        })

        if(value !== '' && value.length > 2) {
            this.setState({
                isFetching: true,
            });

            const DATA_SERVICE_URL = `http://www.omdbapi.com/?apikey=b36c7394&s=${value}`;

            fetch(DATA_SERVICE_URL)
                .then(response => response.json())
                .then(result => {
                    this.setState({
                        results: result.Search ? result.Search : [],
                        isFetching: false,
                        inputValue: value,
                    })
                    console.log('*** result ***');
                    console.log(result);
                })
                .catch(e => {
                    console.log(e);
                    this.setState({
                        isFetching: false
                    });
                });
        }
    }

    render() {
        return (
            <View
                {...this.props}
                {...this.state}
                displayResults={!!this.isResultsEmpty()}
                searchResults={this.searchResults}
                getHighlightedItem={this.getHighlightedItem}
            />
        );
    }
}

LiveInputSearch.propTypes = {
    label: PropTypes.string,
    placeholder: PropTypes.string,
    translations: PropTypes.object,
};

LiveInputSearch.defaultProps = {
    label: '',
    placeholder: '',
    translations: {},
};
