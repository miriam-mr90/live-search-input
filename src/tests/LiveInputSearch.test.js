import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LiveInputSearch from '../LiveInputSearch.js';
import data from '../assets/fixtures';

configure({ adapter: new Adapter() });

describe('Snapshoot test', () => {

    it('renders component', () => {
        const wrapper = shallow(<LiveInputSearch {...data}/>);

        expect(wrapper).toMatchSnapshot();
    });
});
