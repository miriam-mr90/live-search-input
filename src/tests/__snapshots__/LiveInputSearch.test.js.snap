// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshoot test renders component 1`] = `ShallowWrapper {}`;
